Pinta.Core/Effects/BaseEffect.cs

```diff
@@ -84,35 +84,12 @@ public abstract class BaseEffect
 	/// Launches the configuration dialog for this effect/adjustment.
 	/// If IsConfigurable is true, the ConfigDialogResponse event will be invoked when the user accepts or cancels the dialog.
 	/// </summary>
-	public virtual void LaunchConfiguration ()
+	public virtual Task<Gtk.ResponseType> LaunchConfiguration ()
 	{
 		if (IsConfigurable)
 			throw new NotImplementedException ($"{GetType ()} is marked as configurable, but has not implemented LaunchConfiguration");
-	}
-
-	/// <summary>
-	/// Launches the standard configuration dialog for this effect.
-	/// </summary>
-	/// <param name="localizer">
-	/// The localizer for the effect add-in. This is used to fetch translations for the
-	/// strings in the dialog.
-	/// </param>
-	protected void LaunchSimpleEffectDialog (AddinLocalizer localizer)
```

Comment:
I'm pretty sure this signature (or some equivalent way of doing this) is still needed for addins, since this wrapper translates the `Mono.Addins.AddinLocalizer` into the `IAddinLocalizer` interface
Addins can't create an `AddinLocalizerWrapper` manually either since that's an internal class.

e.g. https://github.com/PintaProject/NightVisionEffect/blob/a9ac90a38373cec4c5f69d042998c4ffe35e6438/NightVisionAddin/NightVisionEffect.cs#L53
and 
168115e75d0cc05338d18354439d22e518ffb00b

Reply:
Thank you. I've re-added it, but the return type is a `Task`. Is that fine?

Reply:
Yeah changing the return type to `Task<bool>` is fine - all the demo add-ins will need to be updated for any API changes before release

Reply:
@cameronwhite I know that you are referring to API changes in general (in all the extensible parts of Pinta). With that being said, is there another demo _effect_ besides the night vision one?

Reply:
No, I think that's the only demo for an effect